app/main.py

```diff
@@ -1,52 +1,91 @@
         self.content = content
 
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+
+class BookDisplay:
+    def display(self, book: Book) -> None:
+        raise NotImplementedError
+
+
+class ConsoleBookDisplay(BookDisplay):
+    def display(self, book: Book) -> None:
+        print(book.content)
+
+
+class ReverseBookDisplay(BookDisplay):
+    def display(self, book: Book) -> None:
+        print(book.content[::-1])
+
+
+class BookPrinter:
+    def print_book(self, book: Book) -> None:
+        raise NotImplementedError
+
+
+class ConsoleBookPrinter(BookPrinter):
+    def print_book(self, book: Book) -> None:
+        print(f"Printing the book: {book.title}...")
+        print(book.content)
+
+
+class ReverseBookPrinter(BookPrinter):
+    def print_book(self, book: Book) -> None:
+        print(f"Printing the book in reverse: {book.title}...")
+        print(book.content[::-1])
+
+
+class BookSerializer:
+    def serialize(self, book: Book) -> str:
+        raise NotImplementedError
+
+
+class JsonBookSerializer(BookSerializer):
+    def serialize(self, book: Book) -> str:
+        return json.dumps({"title": book.title, "content": book.content})
+
+
+class XmlBookSerializer(BookSerializer):
+    def serialize(self, book: Book) -> str:
+        root = ElTree.Element("book")
+        title = ElTree.SubElement(root, "title")
+        title.text = book.title
+        content = ElTree.SubElement(root, "content")
+        content.text = book.content
+        return ElTree.tostring(root, encoding="unicode")
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            if method_type == "console":
```

Comment:
Hardcoding the method_type limits the flexibility of your application. Ideally, your design should easily accommodate new types of displays, printers, and serializers without changing the core logic of your application.

Follow the Open/Closed Principle (OCP)